plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -692,6 +692,16 @@ public void testCharTrailingSpace()
         assertUpdate("DROP TABLE char_trailing_space");
     }
 
+    @Test
+    public void testOrPredicatePushdown()
+    {
+        assertThat(query("SELECT * FROM nation WHERE nationkey != 3 OR regionkey = 4")).isFullyPushedDown();
+        assertThat(query("SELECT * FROM nation WHERE nationkey != 3 OR regionkey != 4")).isFullyPushedDown();
+        assertThat(query("SELECT * FROM nation WHERE name = 'ALGERIA' OR regionkey = 4")).isFullyPushedDown();
```

Comment:
Does the expression rewrite depend on how predicates are ordered/grouped. I see that if add another case like

```
        assertThat(query("SELECT * FROM nation WHERE (name = 'ALGERIA' AND regionkey != 10) OR (regionkey = 4 AND name != 'FOOBAR')")).isFullyPushedDown();
```

it fails and we get a ScanFilterNode.

cc: @wendigo

Reply:
That's unrelated to OR. That's because `!=` isn't subsumed:

```
trino:tpch> EXPLAIN SELECT * FROM nation WHERE name != 'FOOBAR' ;
                                                           Query Plan
--------------------------------------------------------------------------------------------------------------------------------
....

 Fragment 1 [SOURCE]
     Output layout: [nationkey, name, regionkey, comment]
     Output partitioning: SINGLE []
     ScanFilter[table = postgresql:tpch.nation tpch.nation, filterPredicate = ("name" <> CAST('FOOBAR' AS varchar(25)))]
         Layout: [nationkey:bigint, name:varchar(25), regionkey:bigint, comment:varchar(152)]
         Estimates: {rows: 25 (3.13kB), cpu: 3.13k, memory: 0B, network: 0B}/{rows: ? (?), cpu: 6.25k, memory: 0B, network: 0B}
         nationkey := nationkey:bigint:int8
         regionkey := regionkey:bigint:int8
         name := name:varchar(25):varchar
         comment := comment:varchar(152):varchar
```